lib/internal/Magento/Framework/Search/Search.php

```diff
@@ -69,7 +69,7 @@ public function search(SearchCriteriaInterface $searchCriteria)
             }
         }
 
-        $this->requestBuilder->setFrom($searchCriteria->getCurrentPage() * $searchCriteria->getPageSize());
+        $this->requestBuilder->setFrom(($searchCriteria->getCurrentPage() > 0 ? $searchCriteria->getCurrentPage() - 1 : 0) * $searchCriteria->getPageSize());
```

Comment:
Hi @chirag-d-wagento. Thank you for your collaboration.

According to the original issue, I'm not sure that changing the `current page` to the `current page - 1` for all cases except page 0 will not bring additional issues. 
Correct me if I'm wrong, but the problem we need to solve here is getting the same results for `current page = 0` and `current page = 1`. So, from my understanding, we need to decrease the `current page` value only in case if the `current page = 1` but not fo all positive values of `current page`

Reply:
Hi @rogyar, I debug this PR. If we decrease the current page value only in case if the **current_page** = 1 we will get next behavior (for example we have 3 products) :

- In case we pass **current_page** 0 and _page_size_ 2, we will get products from 0 to 2
- In case we pass **current_page** 1 and _page_size_ 2, we will get also products from 0 to 2
- In case we pass **current_page** 2 and _page_size_ 2, we will get products from 4 to 6(In our case we will not get our last product)